aredis_om/model/model.py

```diff
@@ -2503,8 +2632,15 @@ async def save(
         self.check()
         db = self._get_db(pipeline)
 
+        # Get model data and apply transformations in the correct order
+        data = self.model_dump()
+        # Convert datetime objects to timestamps for proper indexing
+        data = convert_datetime_to_timestamp(data)
```

Comment:
This allows users to use normal type annotations for datetime fields, instead of a special field type we provide, even if it's messier "on the backend."